core/src/main/java/org/elasticsearch/index/query/TermsQueryParser.java

```diff
@@ -140,14 +108,7 @@ public Query parse(QueryShardContext context) throws IOException, QueryParsingEx
                     throw new QueryParsingException(parseContext, "[terms] query lookup element requires specifying the path");
                 }
             } else if (token.isValue()) {
-                if (EXECUTION_KEY.equals(currentFieldName)) {
-                    // ignore
-                } else if (parseContext.parseFieldMatcher().match(currentFieldName, MIN_SHOULD_MATCH_FIELD)) {
-                    if (minShouldMatch != null) {
-                        throw new IllegalArgumentException("[" + currentFieldName + "] is not allowed in a filter context for the [" + TermsQueryBuilder.NAME + "] query");
-                    }
-                    minShouldMatch = parser.textOrNull();
-                } else if ("boost".equals(currentFieldName)) {
```

Comment:
Please see this comment that answers your notes on deprecated fields: https://github.com/elastic/elasticsearch/pull/12042#issuecomment-130756522 .